### PR TITLE
[various] Add Swift Package Manager integration to example apps

### PR DESCRIPTION
**Pull request by loic-sharma**
This migrates various example apps to have Swift Package Manager integration. This raises the example apps' Flutter SDK requirement to 3.24.0 or higher.

I will get a text exemption for this change.

The example apps were migrated using the following commands:

```
dart run script/tool/bin/flutter_plugin_tools.dart build-examples --ios --swift-package-manager
dart run script/tool/bin/flutter_plugin_tools.dart build-examples --macos --swift-package-manager
```

⚠️ The following plugins' example apps are **not* included in this change as their migration included non-SPM changes that require closer review:

1. google_sign_in
1. google_sign_in_ios
1. image_picker
1. image_picker_ios
1. image_picker_macos
1. pigeon

Part of: https://github.com/flutter/flutter/issues/159173

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran the auto-formatter. (Unlike the flutter/flutter repo, the flutter/packages repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the package surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I [linked to at least one issue that this PR fixes] in the description above.
- [x] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [x] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style], or this PR is [exempt from CHANGELOG changes].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/blob/master/docs/contributing/Tree-hygiene.md
[relevant style guides]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[Discord]: https://github.com/flutter/flutter/blob/master/docs/contributing/Chat.md
[linked to at least one issue that this PR fixes]: https://github.com/flutter/flutter/blob/master/docs/contributing/Tree-hygiene.md#overview
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[exempt from version changes]: https://github.com/flutter/flutter/blob/master/docs/ecosystem/contributing/README.md#version
[following repository CHANGELOG style]: https://github.com/flutter/flutter/blob/master/docs/ecosystem/contributing/README.md#changelog-style
[exempt from CHANGELOG changes]: https://github.com/flutter/flutter/blob/master/docs/ecosystem/contributing/README.md#changelog
[test-exempt]: https://github.com/flutter/flutter/blob/master/docs/contributing/Tree-hygiene.md#tests

### CODE REVIEW

File: packages/in_app_purchase/in_app_purchase/example/ios/Runner.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -247,12 +255,10 @@
 			inputPaths = (
 				"${PODS_ROOT}/Target Support Files/Pods-Runner/Pods-Runner-resources.sh",
 				"${PODS_CONFIGURATION_BUILD_DIR}/in_app_purchase_storekit/in_app_purchase_storekit_privacy.bundle",
-				"${PODS_CONFIGURATION_BUILD_DIR}/shared_preferences_foundation/shared_preferences_foundation_privacy.bundle",
```
Comment (loic-sharma) [inline packages/in_app_purchase/in_app_purchase/example/ios/Runner.xcodeproj/project.pbxproj:250]: This is expected: `in_app_purchase_storekit` hasn't been migrated to SPM yet, but `shared_preferences_foundation` has.

File: packages/local_auth/local_auth_darwin/example/ios/Runner.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -3,7 +3,7 @@
 	archiveVersion = 1;
 	classes = {
 	};
-	objectVersion = 60;
+	objectVersion = 54;
```
Comment (loic-sharma) [inline packages/local_auth/local_auth_darwin/example/ios/Runner.xcodeproj/project.pbxproj:6]: I'm not sure why this was changed. However, this seems reaosonable

File: packages/path_provider/path_provider_foundation/example/ios/Runner.xcodeproj/project.pbxproj [hunk 3]
```diff
@@ -471,7 +461,6 @@
 				CLANG_ENABLE_MODULES = YES;
 				CODE_SIGN_ENTITLEMENTS = Runner/Runner.entitlements;
 				CURRENT_PROJECT_VERSION = "$(FLUTTER_BUILD_NUMBER)";
-				// DEVELOPMENT_TEAM = S8QB4VV633;
```
Comment (loic-sharma) [inline packages/path_provider/path_provider_foundation/example/ios/Runner.xcodeproj/project.pbxproj:474]: FYI this isn't related to SPM but is a reasonable change

File: packages/webview_flutter/webview_flutter_wkwebview/example/ios/Runner.xcodeproj/project.pbxproj [hunk 4]
```diff
@@ -3,14 +3,15 @@
 	archiveVersion = 1;
 	classes = {
 	};
-	objectVersion = 60;
+	objectVersion = 54;
```
Comment (loic-sharma) [inline packages/webview_flutter/webview_flutter_wkwebview/example/ios/Runner.xcodeproj/project.pbxproj:6]: Same comment as https://github.com/flutter/packages/pull/8219#discussion_r1868472445

**Review by tarrinneal — APPROVED** [review]
Shared Prefs approval

**Review by LouiseHsu — APPROVED** [review]
in app purchase ell gee tee em

File: packages/local_auth/local_auth_darwin/example/ios/Runner.xcodeproj/project.pbxproj (hunk 2 above)
Comment (stuartmorgan) [reply packages/local_auth/local_auth_darwin/example/ios/Runner.xcodeproj/project.pbxproj:6]: IIRC this is just based on what version of Xcode was last used to write the project file; it shouldn't matter.

**Comment by stuartmorgan** [comment]
test-exempt: configuration change

**Review by cbracken — APPROVED** [review]
lgtm for google maps!

**Review by stuartmorgan — APPROVED** [review]
LGTM for all packages

**Pull request merged**